test/parallel/test-worker-heap-snapshot.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+// Flags: --expose-internals
+const common = require('../common');
+const assert = require('assert');
+const { Worker } = require('worker_threads');
+const { HeapSnapshotStream } = require('internal/heap_utils');
+const { kHandle } = require('internal/worker');
+
+const worker = new Worker(__filename);
```

Comment:
Maybe I'm missing something -- doesn't this recursively create new worker threads without a depth limit? If you need a dummy worker thread, you can use something like `new Worker('setInterval(() => {}, 100)', { eval: true })` 🙂

Reply:
Uuuuh nice I’m gonna update it!

Reply:
done!